tests/integration/ha/test_ha.py

```diff
@@ -37,8 +75,70 @@ async def test_build_and_deploy(ops_test: OpsTest, kafka_charm):
         assert ops_test.model.applications[APP_NAME].status == "active"
         assert ops_test.model.applications[ZK_NAME].status == "active"
 
+    await ops_test.model.add_relation(APP_NAME, f"{DUMMY_NAME}:{REL_NAME_ADMIN}")
+    await ops_test.model.wait_for_idle(apps=[APP_NAME, DUMMY_NAME], idle_period=30)
+    assert ops_test.model.applications[APP_NAME].status == "active"
+    assert ops_test.model.applications[DUMMY_NAME].status == "active"
+
+
+async def test_replicated_events(ops_test: OpsTest):
+    await ops_test.model.applications[APP_NAME].add_units(count=2)
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", timeout=600, idle_period=120, wait_for_exact_units=3
+    )
+    logger.info("Producing messages and checking on all units")
+    produce_and_check_logs(
+        model_full_name=ops_test.model_full_name,
+        kafka_unit_name=f"{APP_NAME}/0",
+        provider_unit_name=f"{DUMMY_NAME}/0",
+        topic="replicated-topic",
+        replication_factor=3,
+        num_partitions=1,
+    )
+    # check offsets in the two remaining units
+    assert await get_topic_offsets(
```

Comment:
If my understanding is correct, this would check the offset in all units, correct?

Reply:
It's a leftover comment. Since there is only one partition per topic, that's replicated 3 times, checking offsets already gives the numbers for the partition, which apply to all units.